flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/tieredstore/downstream/TieredStoreReader.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.tieredstore.downstream;
+
+import org.apache.flink.runtime.io.network.partition.consumer.InputChannel;
+import org.apache.flink.runtime.io.network.partition.consumer.InputChannel.BufferAndAvailability;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+/** The interface of {@link TieredStoreReader} in Tiered Store. */
+public interface TieredStoreReader {
+
+    /**
+     * Set up the TieredStoreReader.
+     *
+     * @param inputChannels indicate the all producers it consumes.
+     * @param queueChannelReceiver receives a channel and enqueue it.
+     */
+    void setup(InputChannel[] inputChannels, Consumer<InputChannel> queueChannelReceiver);
```

Comment:
UUIC, This `Consumer<InputChannel>` can be passing by constructor instead of lazy `setup`.

Reply:
Done.